Sources/KSCrashRecording/KSCrashConfiguration.m

```diff
@@ -0,0 +1,128 @@
+//
+//  KSCrashConfiguration.m
+//
+//  Created by Gleb Linnik on 11.06.2024.
+//
+//  Copyright (c) 2012 Karl Stenerud. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall remain in place
+// in this source code.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#import "KSCrashConfiguration.h"
+#import <objc/runtime.h>
+#import "KSCrashConfiguration+Private.h"
+
+@implementation KSCrashConfiguration
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self)
+    {
+        _monitors = KSCrashMonitorTypeProductionSafeMinimal;
+        _userInfoJSON = nil;
+        _deadlockWatchdogInterval = 0.0;
+        _enableQueueNameSearch = NO;
+        _enableMemoryIntrospection = NO;
+        _doNotIntrospectClasses = nil;
+        _crashNotifyCallback = nil;
+        _reportWrittenCallback = nil;
+        _addConsoleLogToReport = NO;
+        _printPreviousLogOnStartup = NO;
+        _maxReportCount = 5;
```

Comment:
As this is a wrapper for C struct, maybe it's better to use defaults defined there (e.g. by creating a default struct here)? As it's an easy way to get inconsistency later. And sure I'm not asking reverse mapping of arrays (that are empty) and wrapping callbacks. Having several constants for non-zero values would also work (at least suggest editing in multiple places later)